lib/bolt/plugin/env_var.rb

```diff
@@ -31,7 +32,8 @@ def validate_resolve_reference(opts)
       end
 
       def resolve_reference(opts)
-        ENV[opts['var']] || opts['default']
+        reference = ENV[opts['var']] || opts['default']
+        opts['parse-as-json'] && reference ? JSON.parse(reference) : reference
```

Comment:
This should have a rescue for `JSON::ParserError` so we can raise a nicer message.